src/twisted/web/microdom.py

```diff
@@ -422,16 +421,16 @@ def cloneNode(self, deep=0, parent=None):
 
     def writexml(self, stream, indent='', addindent='', newl='', strip=0,
                  nsprefixes={}, namespace=''):
-        stream.write("<![CDATA[")
-        stream.write(self.nodeValue)
-        stream.write("]]>")
+        stream.write(u"<![CDATA[")
+        stream.write(u"" + self.nodeValue)
+        stream.write(u"]]>")
 
 def _genprefix():
     i = 0
     while True:
         yield  'p' + str(i)
         i = i + 1
-genprefix = _genprefix().next
+genprefix = next(_genprefix())
```

Comment:
this likely means the part that uses genprefix isn't exercised in tests, as this would be a TypeError then
it is called here: https://github.com/twisted/twisted/blob/c94a97f6b5c8bb7bd8392cf086adbc79218fda1c/src/twisted/web/microdom.py#L632
i think we can have `genprefix = _genprefix()` and change the calling line to `prefix = next(genprefix)`